compression/lempel_ziv.py

```diff
@@ -0,0 +1,120 @@
+"""
+    One of the several implementations of Lempel–Ziv–Welch compression algorithm
+    https://en.wikipedia.org/wiki/Lempel%E2%80%93Ziv%E2%80%93Welch
+"""
+
+import os
+import sys
+import math
+
+
+def read_file_binary(file):
```

Comment:
```suggestion
def read_file_binary(file_path: str) -> str:
```
Add type hints as discussed in CONTRIBUTING.md.
Variable name should make it clear to the caller if we are looking for a file name or a file handle.

Are we returning bytes or str?!?.  Remember in Python 2 `bytes == str` but in Python 3 `bytes != str`